mongo/src/org/immutables/mongo/repository/RepositorySetup.java

```diff
@@ -55,13 +63,46 @@
 public final class RepositorySetup {
 
   final ListeningExecutorService executor;
-  final Gson gson;
+  final CodecRegistry codecRegistry;
   final MongoDatabase database;
+  final FieldNamingStrategy fieldNamingStrategy;
 
-  private RepositorySetup(ListeningExecutorService executor, MongoDatabase database, Gson gson) {
+  private RepositorySetup(ListeningExecutorService executor, MongoDatabase database,
+                          CodecRegistry codecRegistry,
+                          FieldNamingStrategy fieldNamingStrategy) {
     this.executor = executor;
     this.database = database;
-    this.gson = gson;
+    this.codecRegistry = codecRegistry;
+    this.fieldNamingStrategy = Preconditions.checkNotNull(fieldNamingStrategy, "fieldNamingStrategy");
+  }
+
+  /**
+   * Allows to customize how attributes are named in mongo queries (this is different from document
```

Comment:
I maybe still don't deep enough here to understand all implications, just sharing some observation: if we cannot align them automatically, is there any chance to explain it better in the javadoc. At first I thought there could be some kind of hack technique to align (field naming in query and in document) automatically. As a late though (I know we initially discussed it, but just in case), would it make any sense if translateName would belong to codec as a strong reminder that it should be implemented in coded to be auto-aligned between documents and queries? thoughts?

Reply:
oh, I see codec registry is coming from driver. Even don't know... Maybe just document it and put field naming strategy as overload to add codec

```
.codecRegistry(registry, namingStrategy)
.codecRegistry(registry) // <-- use default naming strategy
```